larray_editor/editor.py

```diff
@@ -345,21 +337,19 @@ def setup_menu_bar(self):
         help_menu.addAction(create_action(self, _('&About'), triggered=self.about))
 
     def push_changes(self, changes):
-        for key, (old_value, new_value) in changes.items():
-            self.edit_undo_stack.push(EditArrayCommand(self, self.current_array_name, key, old_value, new_value))
-            if logger.isEnabledFor(logging.DEBUG):
-                logger.debug("Edit command pushed: {}[{}] = {}".format(self.current_array_name, key, new_value))
+        self.edit_undo_stack.push(EditArrayCommand(self, self.current_array_name, changes))
 
     def data_changed(self):
-        # We do not set self._unsaved_modifications to True because if users click on `Discard` button
-        # (which calls reject_changes) or choose to display another array, all temporary changes are lost.
-        # `update_title` relies on _is_unsaved_modifications() which checks both self._unsaved_modifications
-        # and self.arraywidget.dirty
+        # We do not set self._unsaved_modifications to True because if users choose to display another array,
+        # all temporary changes are lost. `update_title` relies on self.unsaved_modifications.
         self.update_title()
 
     @property
     def unsaved_modifications(self):
-        return self._unsaved_modifications
+        if self.arraywidget.readonly:
+            return False
```

Comment:
I would not have a special case for readonly here. If we somehow allow editing a readonly array we should let the user know, not try to hide it because hiding it makes the situation worse, not better.